scipy/stats/tests/test_distributions.py

```diff
@@ -1799,6 +1799,73 @@ def test_fit_simple_non_random_data(self):
+    def test_fit_shape(self, a):
+        data = stats.powerlaw.rvs(size=1000, a=a)
+        # obtain the unpenalized log-likelihood function
+
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # compare the results of the LL
+        # a warning is raised when data is generated in some situations, so
+        # it will be supressed.
+        with np.errstate(divide='ignore'):
+            _assert_less_or_close_loglike(stats.powerlaw, data, ll)
+
+    @pytest.mark.parametrize("a", [.5, 1, 1.5])
+    @pytest.mark.parametrize("loc", [0, 1])
+    @pytest.mark.parametrize("scale", [1, 5])
+    @pytest.mark.parametrize("fa", [True, False])
+    @pytest.mark.parametrize("floc", [True, False])
+    @pytest.mark.parametrize("fscale", [True, False])
+    @pytest.mark.parametrize("seed", [1412304, 8435792])
+    def test_fit_MLE_comp_optimzer_combinations(self, a, loc, scale, fa, floc,
+                                                fscale, seed):
+        # the two parametrized seeds exhibit failures for these two previous
+        # issue cases:
+        #  - analytical a = 1 when numerical a > 1
+        #  - analytical a = 1 when numerical a < 1
+        # after inspection this can be moved to setup_method
+        np.random.seed(seed)
+        # skip the test if all parameters are fixed.
+        if False not in {fa, floc, fscale}:
+            pytest.skip()
+        data = stats.powerlaw.rvs(size=250, a=a, loc=loc, scale=scale)
+
+        def ll(args, data):
+            # to get the same behavior as `powerlaw.reduce_func`, negate the
+            # log-likelihood function
+            return -np.sum(np.log(stats.powerlaw.pdf(data, *args)), axis=0)
+
+        # build kwds based on which arguments are indicated to be fixed
+        kwds = dict()
+        if fa:
+            kwds['f0'] = a
+        if floc:
+            kwds['floc'] = loc
+        if fscale:
+            kwds['fscale'] = scale
+        # warnings are raised in the super fit method, unrelated to changes
+        # made here
+        with np.errstate(divide='ignore'):
+            try:
+                _assert_less_or_close_loglike(stats.powerlaw, data, ll, **kwds)
+            except Exception as e:
+                # if the fit is not determined to be better than that obtained
+                # numerically, but the analytically determined `a` is 1and
+                # the numerically determined `a` is slightly less than 1,
+                # this should still be considered a good fit.
+                a_analytical = stats.powerlaw.fit(data, **kwds)[0]
+                a_numerical = super(type(stats.powerlaw),
+                                    stats.powerlaw).fit(data, **kwds)[0]
+                if not (a_analytical == 1 and .9 < a_numerical < 1):
+                    raise e
```

Comment:
@mdhaber We discussed this scenario yesterday on zoom, and I think that you said that this scenario was ok:

analytical result of a = 1
numerical result a slightly less than 1. 

Correct?

Reply:
Yes. I'm not sure if we need the > 0.9 part in SciPy ultimately, but it's ok to leave it right now. I'll write this all up soon, I hope.